app/controllers/articles_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
        
     private
 
+    def check_subscriber
+        if Article.find(params[:id]).premium? == true 
+            if user_signed_in? && current_user.subscriber?
```

Comment:
I think this should be enough`if current_user.subscriber?` because there is no current_user unless the user is signed in.

Reply:
I think we need both. If a visitor clicks and are not signed in. the conditional current_user.subscriber? would generate an error because current_user is a Nil class. 

If we keep the first one as well it will handle the visitors trying to access the content.